src/test/relays/BasicRelay.t.sol

```diff
@@ -270,6 +270,50 @@ contract BasicRelayTest is DSTest {
         assertEq(dai.balanceOf(feeCollector), 1 ether);
     }
 
+    function test_relay_no_fee_collector() public {
+        _whitelistThis();
+        uint256 sk = uint256(keccak256(abi.encode(8)));
+        address receiver = hevm.addr(sk);
+        TeleportGUID memory guid = TeleportGUID({
+            sourceDomain: "l2network",
+            targetDomain: "ethereum",
+            receiver: addressToBytes32(receiver),
+            operator: addressToBytes32(address(relay)),
+            amount: 100 ether,
+            nonce: 5,
+            timestamp: uint48(block.timestamp)
+        });
+        uint256 maxFeePercentage = WAD * 1 / 100;   // 1%
+        uint256 gasFee = WAD;                       // 1 DAI of gas
+        uint256 expiry = block.timestamp;
+        bytes32 signHash = getSignHash(
+            guid,
+            maxFeePercentage,
+            gasFee,
+            expiry
+        );
+
+        (uint8 v, bytes32 r, bytes32 s) = hevm.sign(sk, signHash);
+
+        assertEq(dai.balanceOf(receiver), 0);
+
+        // relay() should succeed even without the appended feeCollector
+        // but the fee will be sent to an incorrect address
```

Comment:
I guess it will always go to `address(0)`, right?

Reply:
I think in general it will go to a non-zero junk address derived from the last bytes of the last signature (i.e. the end of the calldata)